src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1170,9 +1170,9 @@ internal static long DoubleDateToTicks(double value)
         //
         public override bool Equals([NotNullWhen(true)] object? value)
         {
-            if (value is DateTime)
+            if (value is DateTime time)
             {
-                return Ticks == ((DateTime)value).Ticks;
+                return this == time;
             }
             return false;
```

Comment:
Can be simplified to just
```csharp
public override bool Equals([NotNullWhen(true)] object? value) =>
    value is DateTime dt && dt == this;
```
🙂

Reply:
👍 all done, verified no codegen changes. [Didn't expect any of course, but just in case 😉]